src/pages/namespace/Monitoring/Logs/Scrollcontainer.tsx

```diff
@@ -0,0 +1,145 @@
+import { useEffect, useRef, useState } from "react";
+
+import { ArrowDown } from "lucide-react";
+import Button from "~/design/Button";
+import Entry from "./Entry";
+import { Logs } from "~/design/Logs";
+import { twMergeClsx } from "~/util/helpers";
+import { useLogsPreferencesWordWrap } from "~/util/store/logs";
+import { useNamespacelogs } from "~/api/namespaces/query/logs";
+import { useTranslation } from "react-i18next";
+import { useVirtualizer } from "@tanstack/react-virtual";
+
+const ScrollContainer = () => {
+  const wordWrap = useLogsPreferencesWordWrap();
+
+  const { t } = useTranslation();
+
+  const { data: logData } = useNamespacelogs();
+  const [watch, setWatch] = useState(true);
+
+  // The scrollable element for the list
+  const parentRef = useRef<HTMLDivElement | null>(null);
+
+  // The virtualizer
+  const rowVirtualizer = useVirtualizer({
+    count: logData?.results.length ?? 0,
+    getScrollElement: () => parentRef.current,
+    estimateSize: () => 20,
+    /**
+     * overscan is the number of items to render above and below
+     * the visible window. More items = less flickering when
+     * scrolling, but more memory usage and initial load time.
+     * I tested it with around 2000 items and 40 was a good fit
+     * to have no flickering with pretty high scrolling speed.
+     */
+    overscan: 40,
+    getItemKey: (index) => logData?.results[index]?.t ?? index,
+  });
+
+  useEffect(() => {
+    if (logData?.results.length && watch) {
+      /**
+       * monkey patch ahead 🙈
+       * calling rowVirtualizer.scrollToIndex(logData?.results.length);
+       * is supposed to scroll to the bottom of the list, but it doesn't
+       * work when the height of the line is dynamic. The calcuation that
+       * is used only takes the result of estimateSize call into account
+       * (we can test that by setting estimateSize very high)
+       *
+       * the only solution that workd for now is calling
+       * rowVirtualizer.scrollElement with a very high top value
+       * two times in a row, with a small delay for one a render
+       * cycle in between
+       */
+      rowVirtualizer.scrollElement?.scrollTo({
+        top: 999999999999999,
+      });
+
+      const timeOut = setTimeout(() => {
+        rowVirtualizer.scrollElement?.scrollTo({
+          top: 999999999999999,
+        });
+      }, 10);
+
+      return () => {
+        clearTimeout(timeOut);
+      };
+    }
+  }, [logData?.results.length, rowVirtualizer, watch]);
+
+  if (!logData) return null;
+
+  const items = rowVirtualizer.getVirtualItems();
+
+  return (
+    <Logs
+      wordWrap={wordWrap}
+      className="h-full overflow-scroll"
+      ref={parentRef}
+      onScroll={(e) => {
+        const element = e.target as HTMLDivElement;
+        if (element) {
+          const { scrollHeight, scrollTop, clientHeight } = element;
+          const scrollDistanceToBottom =
+            scrollHeight - scrollTop - clientHeight;
```

Comment:
I'll mention this in the main PR review comment: There's a bug with scrolling to bottom in Firefox. 

I may be off target, but I have a hunch that somewhere around here something behaves differently in Firefox causing that bug, maybe due to the way it calculates these values?

Speaking of which, I think we should make a ticket for later to refactor the two occurrences of this `Scrollcontainer` into one reusable component.